lib/config/presets/index.spec.ts

```diff
@@ -290,6 +290,36 @@ describe('config/presets/index', () => {
       expect(local.getPreset.mock.calls[0][0].baseConfig).toBeDefined();
       expect(res).toMatchSnapshot();
     });
+
+    it('resolves self-hosted transitive presets without baseConfig', async () => {
+      config.platform = 'gitlab';
+      config.endpoint = 'https://dummy.example.com/api/v4';
+      config.extends = ['local>username/preset-repo'];
+      local.getPreset
+        .mockResolvedValueOnce({
+          extends: ['local>username/preset-repo//subpreset'],
+        })
+        .mockResolvedValueOnce({ labels: ['self-hosted resolved'] });
+
+      const res = await presets.resolveConfigPresets(config);
+
+      expect(res).toEqual({
+        platform: 'gitlab',
+        endpoint: 'https://dummy.example.com/api/v4',
+        labels: ['self-hosted resolved'],
+      });
+      expect(local.getPreset.mock.calls).toHaveLength(2);
+      expect(local.getPreset.mock.calls[0][0].baseConfig).toEqual({
+        platform: 'gitlab',
+        endpoint: 'https://dummy.example.com/api/v4',
+        extends: ['local>username/preset-repo'],
+      });
+      expect(local.getPreset.mock.calls[1][0].baseConfig).toEqual({
+        platform: 'gitlab',
+        endpoint: 'https://dummy.example.com/api/v4',
+        extends: ['local>username/preset-repo'],
+      });
```

Comment:
```suggestion
      expect(local.getPreset.mock.calls).toEqual([
        [
          {
            platform: 'gitlab',
            endpoint: 'https://dummy.example.com/api/v4',
            extends: ['local>username/preset-repo'],
          },
        ],
        [
          {
            platform: 'gitlab',
            endpoint: 'https://dummy.example.com/api/v4',
            extends: ['local>username/preset-repo'],
          },
        ],
      ]);
```

isn't this simpler?

Reply:
This isn't possible as the array contains 3 additional parameters which I chose to ignore as it isn't relevant for the tested behavior.

Reply:
then use `toMatchObject` and add `expect.any` for other array params

Reply:
To be honest, I don't see if this actually increases readability. I don't want to be rude, but I'd rather see that merged fast instead of reiterating over minor improvements to test code for a problem that blocks a documented use case in production.

That being said, I will implement the changes next week after I'm back from vacation if the PR is still open. Also feel free to change the PR as desired - I'd enabled changes for maintainers.